File: Java/src/main/java/com/gildedrose/AgedBrieItem.java
package com.gildedrose;

public class AgedBrieItem extends Item {

    public AgedBrieItem(String name, int sellIn, int quality) {
        super(name, sellIn, quality);
    }

    @Override
    protected void updateQualityBeforeSellIn() {
        increaseQualityIfBelow50();
    }

    @Override
    protected void updateSellIn() {
        decreaseSellIn();
    }

    @Override
    protected void updateQualityAfterSellIn() {
        if (sellIn < 0) {
            increaseQualityIfBelow50();
        }
    }
}

File: Kotlin/src/main/kotlin/com/gildedrose/AgedBrieItem.kt
package com.gildedrose

class AgedBrieItem(override val name: String, override var sellIn: Int, override var quality: Int) : Item(name, sellIn, quality) {

    override fun updateQuality() {
        increaseQuality()
    }

    override fun updateSellIn() {
        decreaseSellIn()
    }

    override fun updateQualityWhenExpired() {
        increaseQuality()
    }
}